src/libraries/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -518,6 +518,9 @@
   <data name="CannotPopulateCollection" xml:space="preserve">
     <value>The collection type '{0}' is abstract, an interface, or is read only, and could not be instantiated and populated.</value>
   </data>
+  <data name="ConstructorContainsNullParameterNames" xml:space="preserve">
+    <value>The deserialization constructor for type '{0}' contains parameters with null names. This might happen because the parameter names have been trimmed by the linker. Consider using the source generated serializer instead.</value>
```

Comment:
```suggestion
    <value>The deserialization constructor for type '{0}' contains parameters with null names. This might happen when the parameter names have been trimmed by the linker. Consider using the source generated serializer instead.</value>
```

Reply:
Trimming happens at compile time, but the use of "when" suggests it is something happening now (i.e. at run time). Compare "because they have been trimmed" with "when they are being trimmed".

Reply:
doesn't sound like that to me but either way, this is non-blocking comment since it's obvious what it means